pyupgrade/_plugins/open_mode.py

```diff
@@ -21,19 +23,35 @@
 U_MODE_REPLACE = U_MODE_REPLACE_R | U_MODE_REMOVE_U
 
 
-def _fix_open_mode(i: int, tokens: List[Token]) -> None:
+class FunctionArg(NamedTuple):
+    arg_idx: int
+    value: ast.expr
+
+
+def _fix_open_mode(i: int, tokens: List[Token], arg_idx: int) -> None:
     j = find_open_paren(tokens, i)
     func_args, end = parse_call_args(tokens, j)
-    mode = tokens_to_src(tokens[slice(*func_args[1])])
-    mode_stripped = mode.strip().strip('"\'')
+    if len(func_args) < 2:
+        return
+    mode = tokens_to_src(tokens[slice(*func_args[arg_idx])])
+    mode_stripped = mode.split('=')[-1]
+    mode_stripped = mode_stripped.strip().strip('"\'')
     if mode_stripped in U_MODE_REMOVE:
-        del tokens[func_args[0][1]:func_args[1][1]]
+        if arg_idx == 0:
+            next_token_slice = slice(*func_args[arg_idx + 1])
+            next_argument = tokens_to_src(tokens[next_token_slice])
+            tokens[next_token_slice] = [
+                Token('SRC', next_argument.lstrip(' ')),
+            ]
```

Comment:
does this do weird stuff for multiline -- my guess is it doesn't come up much in practice -- it's also possible the existing code is broken for that too -- in which case just ignore me

Reply:
I tried this out an it seemed to work fine with multilines - however, it's true that this doesn't come up much in practice (I've never seen it, I was just trying to cover all cases), so I've removed the extra logic as maybe it makes to leave whitespace formatting to other tools?